runtime.js

```diff
@@ -121,8 +119,15 @@
         } else {
           Promise.resolve(info.value).then(callNext, callThrow);
         }
+      };
+
+      if (process && process.domain) {
```

Comment:
This does not ensure that process is declared. It only ensures that process does not have a falsy value. It will still throw if it is not declared. typeof is pretty much the only way to go here.